omod/src/test/java/org/openmrs/module/idgen/rest/controller/AutoGenerationOptionControllerTest.java

```diff
@@ -0,0 +1,208 @@
+	private IdentifierSourceService identifierSourceService;
+	
+	public static final String NEW_LOCATION = "9356400c-a5a2-4532-8f2b-2361b3446eb8";
+	
+	public static final String NEW_SOURCE = "0d47284f-9e9b-4a81-a88b-8bb42bc0a901";
+	
+	public static final String LOCATION = "8d6c993e-c2cc-11de-8d13-0010c6dffd0f";
+	
+	public static final String SOURCE = "0d47284f-9e9b-4a81-a88b-8bb42bc0a902";
+	
+	public static final String AUTO_GENERATION_OPTION_UUID = "2";
+	
+	public static final String IDENTIFIER_TYPE_UUID = "2f470aa8-1d73-43b7-81b5-01f0c0dfa53c";
+	
+	public static final String UNKNOWN_LOCATION = "8d6c993e-c2cc-11de-8d13-0010c6dffd0f";
+	
+	@Before
+	public void before() throws Exception {
+		executeDataSet("org/openmrs/module/idgen/include/TestData.xml");
+	}
+	
+	@Override
+	public String getURI() {
+		return "idgen/autogenerationoption";
+	}
+	
+	@Override
+	public String getUuid() {
+		return AUTO_GENERATION_OPTION_UUID;
+	}
+	
+	@Override
+	public long getAllCount() {
+		Integer allAutoGenerationOptions = 0;
+		for (PatientIdentifierType patientIdentifierType : Context.getPatientService().getAllPatientIdentifierTypes()) {
+			allAutoGenerationOptions += identifierSourceService.getAutoGenerationOptions(patientIdentifierType).size();
+		}
+		return allAutoGenerationOptions;
+	}
+	
+	@Test()
+	public void shouldCreateNewAutoGenerationOption() throws Exception {
+		long originalCount = getAllCount();
+		boolean automaticGenerationEnabled = true;
+		boolean manualEntryEnabled = true;
+		SimpleObject autoGenerationOption = new SimpleObject();
+		autoGenerationOption.add("identifierType", IDENTIFIER_TYPE_UUID);
+		autoGenerationOption.add("location", LOCATION);
+		autoGenerationOption.add("source", SOURCE);
+		autoGenerationOption.add("automaticGenerationEnabled", automaticGenerationEnabled);
+		autoGenerationOption.add("manualEntryEnabled", manualEntryEnabled);
+		String json = new ObjectMapper().writeValueAsString(autoGenerationOption);
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI());
+		req.setContent(json.getBytes());
+		SimpleObject newAutoGenerationOption = deserialize(handle(req));
+		assertNotNull(PropertyUtils.getProperty(newAutoGenerationOption, "uuid"));
+		assertEquals(originalCount + 1, getAllCount());
+		assertEquals(SOURCE, PropertyUtils.getProperty(newAutoGenerationOption, "source"));
+		assertEquals(LOCATION, PropertyUtils.getProperty(newAutoGenerationOption, "location"));
+		assertEquals(IDENTIFIER_TYPE_UUID, PropertyUtils.getProperty(newAutoGenerationOption, "identifierType"));
+		assertEquals(automaticGenerationEnabled,
+		    PropertyUtils.getProperty(newAutoGenerationOption, "automaticGenerationEnabled"));
+		assertEquals(manualEntryEnabled, PropertyUtils.getProperty(newAutoGenerationOption, "manualEntryEnabled"));
+	}
+	
+	@Test()
+	public void shouldCreateNewAutoGenerationOptionWithoutLocation() throws Exception {
+		long originalCount = getAllCount();
+		boolean automaticGenerationEnabled = true;
```

Comment:
What is the use of these two local variables? Can't you just pass in the actual values?